src/mscorlib/src/System/Runtime/CompilerServices/jithelpers.cs

```diff
@@ -216,14 +216,7 @@ static internal PinningHelper GetPinningHelper(Object o)
         private extern static bool IsAddressInStack(IntPtr ptr);
 #endif
 
-        static internal bool ByRefLessThan<T>(ref T refA, ref T refB)
-        {
-            // The body of this function will be replaced by the EE with unsafe code!!!
-            // See getILIntrinsicImplementation for how this happens.
-            throw new InvalidOperationException();
-        }
-
-        static internal ref T GetArrayData<T>(T[] array)
+        static internal ref byte GetRawSzArrayData(this Array array)
```

Comment:
METHOD__JIT_HELPERS__GET_ARRAY_DATA and 
METHOD__JIT_HELPERS__GET_RAW_SZ_ARRAY_DATA case both point to the same [code](https://github.com/dotnet/coreclr/blob/master/src/vm/jitinterface.cpp#L6935-L6954) and yet one returns a ref T while the other returns a ref byte? How does that work?